### ISSUE TITLE
Different Qr Code on different Pcs

### ISSUE THREAD

**Issue by Abooow786** [zxing-cpp/zxing-cpp#727]
Hey Axxel,

sorry that i didn't answer on the last issue with the same topic.. There happened some things and i couldn't respond. I hope you can forgive me this...

To come back to the Problem. Maybe it interests you or somebody else. I went back to the Windows7 PC installed the remote Debugger, searched for all needed Dlls to start the Debug exe and .... Yeah took some time.

The finale Problem was following. In the File "BitHacks.h" function "NumberOfLeadingZeros" you work with "_lzcnt". This function returned some random integers.. After an quick look up at:
[https://learn.microsoft.com/de-de/cpp/intrinsics/lzcnt16-lzcnt-lzcnt64?view=msvc-170](url)
i realised that i need the Haswell Architecture. Sooo it will just return random integers and surely my qr code will be wrong:). Because of the other functions which embed the fixed positioning i'm assuming that the wrong qrCode was still somewhere near the right one?
So there are 2 possible Solutions for this. Check if the Hardware supports it if not return an error or writing a different function to get the leading Zeros.
[https://learn.microsoft.com/de-de/cpp/intrinsics/bitscanreverse-bitscanreverse64?view=msvc-170](url)
This function does something similar and works on every hardware just fine:) So more people and more Pcs could use ZXing:)
I will pull a request with a quick & dirty solution so you can maybe take a better look at this.
If you or somebody else has an question feel free to ask

**Comment by axxel** [comment]
Wow, that is wild. Thanks for all the debugging effort. A function that returns random numbers on random CPUs - who would have thought. Your fix seems pretty much in line with the existing implementation of the 'forward' version a few lines below your change in the same file. Looking at it, I noticed though that you missed the 64-bit version and I discovered another issue with the existing implementation. So I went ahead and fixed all of that myself.
